components/ModalRoot.vue

```diff
@@ -0,0 +1,56 @@
+<template>
+  <SfModal class="sf-modal" :visible="!!componentRef" @close="handleClose">
+    <template #modal-bar>
+      <SfBar class="sf-modal__bar bar-heading" :title="titleRef" @click:close="handleClose" />
+    </template>
+    <component :is="componentRef" :properties="properties" @onClose="handleClose" />
+  </SfModal>
+</template>
+<script lang="ts">
+import { SfModal, SfSearchBar, SfBar } from "@storefront-ui/vue"
+import { ref } from "@vue/composition-api"
+import { ModalBus } from "../eventBus/eventBus"
+export default {
+  components: { SfModal, SfSearchBar, SfBar },
+  setup() {
+    const componentRef = ref(null)
+    const titleRef = ref("")
+    const properties = ref(null)
+
+    const handleClose = () => {
+      componentRef.value = null
+    }
+
+    const getTitle = (componentName) => {
+      const titles = {
+        StoreLocatorModal: "Select Store",
```

Comment:
This is still hardcoded, it needs to be passed in as a prop or with the Store Locator Modal.

Reply:
Passed as a prop from the components.